stock_release_channel_partner_public_holidays/models/stock_release_channel.py

```diff
@@ -0,0 +1,54 @@
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+from odoo import fields, models
+
+
+class StockReleaseChannel(models.Model):
+
+    _inherit = "stock.release.channel"
+
+    exclude_public_holidays = fields.Boolean()
+
+    def filter_release_channel(self, partner):
+        channels = self
+        for channel in self:
+            if not channel.exclude_public_holidays:
+                continue
+            if channel._is_shipment_date_a_public_holiday(partner):
+                channels -= channel
+        return channels
+
+    def _is_shipment_date_a_public_holiday(self, partner):
+        """
+        Returns True if shipment_date is a public holiday
+        :return: bool
+        """
+        self.ensure_one()
+        res = False
+        shipment_date = self.shipment_date
+        if not shipment_date:
+            return res
+        holidays_filter = [("year", "=", shipment_date.year)]
+        if partner:
+            if partner.country_id:
+                holidays_filter.append("|")
+                holidays_filter.append(("country_id", "=", False))
+                holidays_filter.append(("country_id", "=", partner.country_id.id))
```

Comment:
FYI, you can write this as `holidays_filter.append(("country_id", "in", (False, partner.country_id.id)))`

Reply:
This should even work if there is no country on the partner as it will resolve as `(False, False)`, allowing to drop the if/else

Reply:
`("country_id", "in", (False, False))` is not good, it's against the indexes search.

Reply:
https://dev.to/yugabyte/postgresql-do-different-equality-predicates-make-a-difference-1c7

Reply:
there is no different between `("country_id", "in", [False])` and `("country_id", "=", [False])`, because DMS is smart enough to convert "in" into "=". But `("country_id", "in", (False, False))` is not.